meson.build

```diff
@@ -29,29 +33,43 @@ else
     endif
 endif
 
-curl_dep = dependency('libcurl')
-
-if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
-    argplib = cc.find_library('argp')
+if build_machine.system() == 'windows'
+    curl_dep = dependency('curl', modules: 'CURL::libcurl')
 else
-    argplib = dependency('', required : false)
+    curl_dep = dependency('libcurl')
+endif
+
+if build_machine.system() != 'windows'
+    if build_machine.system() == 'darwin' or build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); return 0; }; void main() {}')
+       argplib = cc.find_library('argp')
+    else
+       argplib = dependency('', required : false)
+    endif
 endif
 
 if get_option('with-openssl') == 'disabled'
     openssl_dep = dependency('', required : false)
 else
-    openssl_dep = dependency('openssl', required : get_option('with-openssl') == 'enabled')
+    if build_machine.system() == 'windows'
+        openssl_dep = dependency('openssl', modules : ['OpenSSL::SSL', 'OpenSSL::Crypto'], required : get_option('with-openssl') == 'enabled')
+    else
+        openssl_dep = dependency('openssl', required : get_option('with-openssl') == 'enabled')
+    endif
     if openssl_dep.found()
         add_project_arguments('-DZCHUNK_OPENSSL', language : 'c')
     endif
 endif
 
-inc = include_directories('include')
+inc = []
+inc += include_directories('include')
+if host_machine.system() == 'windows'
+    inc += include_directories('src/lib/win32')
+endif
 
 subdir('include')
 subdir('src')
-if not get_option('coverity')
-    subdir('test')
+if not get_option('coverity') and host_machine.system() != 'windows'
```

Comment:
I really, really don't like this.  How do we know zchunk is working correctly without these tests?

Reply:
Well, I've put in some additional work to get argp to compile on Windows and it looks like I can make it work. I've got 70% of the programs in zchunk compiled now. Will try to get to the rest today and run the tests.

Otherwise, we can always test in downstream projects like powerloader which are cross-platform from the get-go. C++ makes cross-platform easier and there are a bunch of different libraries for CLI handling that work the same on all three platforms.

Reply:
With the GitHub actions added for Ubuntu and MacOS, are you good to add Windows support here?

Reply:
yep, I just pushed a commit that tries to run the compilation (not tests!) for Windows in the Github actions.